### PR TITLE
Create 'legacy' style keys for the Thunderbird sync scope

### PR DESCRIPTION
**Pull request by darktrojan**

### CODE REVIEW

File: test/deriver/ScopedKeys.js [hunk 1]
```diff
@@ -70,6 +70,21 @@ describe('ScopedKeys', function () {
       });
   });
 
+  it('should correctly derive Thunderbird sync key to known test vectors', () => {
```
Comment (darktrojan) [inline test/deriver/ScopedKeys.js:73]: These are the same values as the test for the `oldsync` key. Does it matter?

**Comment by clouserw** [comment]
I wonder why the tests didn't run..

**Comment by darktrojan** [comment]
That's better. Looks like `build_and_test` doesn't actually include building.

**Comment by darktrojan** [comment]
Could somebody give me an ETA on this reaching the staging server?

**Comment by clouserw** [comment]
Tracking @ https://mozilla-hub.atlassian.net/browse/FXA-9691 to get on the roadmap

File: test/deriver/ScopedKeys.js (hunk 1 above)
Comment (bbangert) [reply test/deriver/ScopedKeys.js:73]: I don't believe it should.

**Review by bbangert — APPROVED** [review]
Looks fine to me, given my limited understanding of this. 😉

**Review by vbudhram — COMMENTED** [review]
@darktrojan Hello, the code here looks good. Is there a bug or ticket I could reference on why this change was needed? I'm not familiar with how Thunderbird used Sync.

**Comment by darktrojan** [comment]
Thunderbird is getting a separate client ID and keys so that it cannot decrypt Firefox data, and Firefox cannot decrypt Thunderbird's data. We're making the change here so that both clients behave the same way (same length keys, etc.).

**Review by vbudhram — APPROVED** [review]
@darktrojan Thanks for the update, I think this is good to merge.

**Pull request merged**